posts/setting-up-a-shared-library-and-seed-job-in-jenkins-1.md

```diff
@@ -0,0 +1,134 @@
+
+## Part 1 Goals
+1. Set up a Jenkins Freestyle Job `called seedJob` to create and configure other jobs using the Jenkins Job DSL API.
+2. Store the `seedJob` configuration in github.
+3. Run the `seedJob` to create a Freestyle Job called `poc-micro`. 
+
+## Prerequisites
+1. **Docker installed** 
+   * If you do not have docker installed yet, proceed to Docker's [Getting Started](https://docs.docker.com/get-started/) guide first.
+2. **Maven ready Jenkins running in a Docker container**
+    * Pull the Jenkins image: `docker pull kcrane121/maven-jenkins:blog`.
+        * This image is based on the [Jenkins Official Repository](https://hub.docker.com/_/jenkins/).
+    * Start the Jenkins container: `docker run -p 8080:8080 -p 50000:50000 kcrane121/maven-jenkins:blog`.
+    * Open a browser and navigate to `http://localhost:8080/` and follow the instructions to complete the set up.
+      **Set up Tips:**
+      1. The first time you set this up, you will need to provide a password that was given to you after running `docker run -p 8080:8080 -p 50000:50000 kcrane121/maven-jenkins:blog`.  
+         * The generated admin password can also be foun at `/var/jenkins_home/secrets/initialAdminPassword` in the docker container.
+      3. Select `Install Suggested Plugins` during the setup process .
+3. **Familiarity with [Jenkins Pipeline](https://blog.ippon.tech/continuous-delivery-with-jenkins-pipeline/)**.
+
+## Goal 1
+Now that the prerequisites are out of the way, the first thing we are going to do is set up a **Freestyle project** `seedJob` in Jenkins. This job will be used to generate other jobs you want to onboard. 
+
+### Install the `Job DSL` plugin
+Navigate to `http://localhost:8080/` in your browser and login to Jenkins with the credentials you set up (or the default admin ones provided to you during the initial set up). We need to configure Jenkins to use the Jenkins Job DSL API. This provides us the functionality to configure how we want our new jobs built.
+
+  1. Navigate to `Dashboard` > `Manage Jenkins` > `Manage Plugins` > select `Available` tab > search for `Job DSL` and install.
+
+### Creating the Freestyle Project `seedJob`
+We will set up our `seedJob` or `job that creates jobs`. This creation is done the same way any other job is created.
+
+  1. On the left hand side of the page, select `New Item`
+  2. In the text box for `Enter an item name`, enter `seedJob` > select the `Freestyle project` > select `OK`.
+  ![jenkins freestyle project](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/05/jenkins-shared-library-1.1.png)
+
+### Configure `seedJob` to use `microservice-pipelines` repository
+Since we are using the `microservice-pipelines` repository to load `seed.groovy`, we need to configure `seedJob` to use this repository.
```

Comment:
Explain why we need to use this repository? I'm not sure why it's needed in part 1 btw

Reply:
I added an outline about why using Shared Library code becomes useful as the number of jobs maintained grows/changes